components/browser/state/src/main/java/mozilla/components/browser/state/reducer/ContentStateReducer.kt

```diff
@@ -128,6 +129,66 @@ internal object ContentStateReducer {
             is ContentAction.UpdateHistoryStateAction -> updateContentState(state, action.sessionId) {
                 it.copy(history = HistoryState(action.historyList, action.currentIndex))
             }
+            is ContentAction.UpdatePermissionsRequest -> updateContentState(
+                state,
+                action.sessionId
+            ) {
+                if (!it.permissionRequestsList.doesContain(action.permissionRequest)) {
+                    it.copy(
+                        permissionRequestsList = it.permissionRequestsList + action.permissionRequest
```

Comment:
nit: could be do `permissionRequestsList +=action.permissionRequest`? :)

Reply:
The same could apply to other variants :)

Reply:
`permissionRequestList` is actually a named argument there.
 
![image](https://user-images.githubusercontent.com/60002907/96998503-38cb4a00-153c-11eb-8663-e9b4919e1003.png)